pkg/linear-pools/contracts/silo-linear-pool/SiloLinearPool.sol

```diff
@@ -61,10 +59,8 @@ contract SiloLinearPool is LinearPool, Version {
         Version(args.version)
     {
         _shareToken = IShareToken(address(args.wrappedToken));
```

Comment:
We should prefer type-casting over redundant storage. I would do the following:

1. Make `IShareToken is IERC20` to reduce some of our casting.
2. Eliminate this `_shareToken` variable.
3. Use `IShareToken(args.wrappedToken)` within the constructor.
4. Use `IShareToken(getWrappedToken())` elsewhere.